pkg/controllers/webhook/controller.go

```diff
@@ -754,6 +762,129 @@ func (c *controller) buildDefaultResourceValidatingWebhookConfiguration(_ contex
 		nil
 }
 
+func scanResourceFilter(resFilter kyvernov1.ResourceFilters, operationStatusMap map[string]bool) (bool, map[string]bool) {
+	opFound := false
+	for _, rf := range resFilter {
+		if rf.ResourceDescription.Operations != nil {
+			for _, o := range rf.ResourceDescription.Operations {
+				opFound = true
+				operationStatusMap[string(o)] = true
+			}
+		}
+	}
+	return opFound, operationStatusMap
+}
+
+func computeOperationsForValidatingWebhookConf(rules []kyvernov1.Rule, operationStatusMap map[string]bool) map[string]bool {
+	for _, r := range rules {
+		opFound := false
+		if len(r.MatchResources.Any) != 0 {
+			opFound, operationStatusMap = scanResourceFilter(r.MatchResources.Any, operationStatusMap)
+		}
+		if len(r.MatchResources.All) != 0 {
+			opFound, operationStatusMap = scanResourceFilter(r.MatchResources.All, operationStatusMap)
+		}
+		if len(r.ExcludeResources.Any) != 0 {
+			opFound, operationStatusMap = scanResourceFilter(r.ExcludeResources.Any, operationStatusMap)
+		}
+		if len(r.ExcludeResources.All) != 0 {
+			opFound, operationStatusMap = scanResourceFilter(r.ExcludeResources.All, operationStatusMap)
+		}
+		if r.MatchResources.ResourceDescription.Operations != nil {
+			for _, o := range r.MatchResources.ResourceDescription.Operations {
+				opFound = true
+				operationStatusMap[string(o)] = true
+			}
+		}
+		if r.ExcludeResources.ResourceDescription.Operations != nil {
+			for _, o := range r.ExcludeResources.ResourceDescription.Operations {
+				opFound = true
+				operationStatusMap[string(o)] = true
+			}
+		}
+		if !opFound {
+			operationStatusMap[webhookCreate] = true
+			operationStatusMap[webhookUpdate] = true
+			operationStatusMap[webhookConnect] = true
+			operationStatusMap[webhookDelete] = true
+		}
+	}
+	return operationStatusMap
+}
+
+func isMutationEmpty(m kyvernov1.Mutation) bool {
+	if len(m.Targets) > 0 {
+		return false
+	}
+	if m.RawPatchStrategicMerge != nil && len(m.RawPatchStrategicMerge.Raw) > 0 {
+		return false
+	}
+	if m.PatchesJSON6902 != "" {
+		return false
+	}
+	if len(m.ForEachMutation) > 0 {
+		return false
+	}
+	return true
+}
+
+func computeOperationsForMutatingWebhookConf(rules []kyvernov1.Rule, operationStatusMap map[string]bool) map[string]bool {
+	for _, r := range rules {
+		if !isMutationEmpty(r.Mutation) || len(r.VerifyImages) > 0 {
```

Comment:
Can `isMutationEmpty(r.Mutation)` be replaced by `rule.HasMutate()`? Same for "VerifyImages" -> `rule.HasVerifyImages()`.

Reply:
Looks like I can, cant recall why I didnt do it using these in the first place. Let me make these changes and try.